currency/pairs.go

```diff
@@ -199,28 +199,26 @@ func (p Pairs) GetPairsByCurrencies(currencies Currencies) Pairs {
 	return pairs
 }
 
-// Remove removes the specified pair from the list of pairs if it exists
-func (p Pairs) Remove(pair Pair) (Pairs, error) {
-	pairs := slices.Clone(p)
-	for x := range p {
-		if p[x].Equal(pair) {
-			return append(pairs[:x], pairs[x+1:]...), nil
+// Remove removes the specified pairs from the list of pairs if they exist
+func (p Pairs) Remove(rem ...Pair) Pairs {
+	n := make(Pairs, 0, len(p))
+	for _, pN := range p {
+		if !slices.ContainsFunc(rem, func(pX Pair) bool { return pX.Equal(pN) }) {
+			n = append(n, pN)
 		}
 	}
-	return nil, fmt.Errorf("%s %w", pair, ErrPairNotFound)
+	return slices.Clip(n)
 }
 
-// Add adds a specified pair to the list of pairs if it doesn't exist
+// Add adds pairs to the list of pairs ignoring duplicates
 func (p Pairs) Add(pairs ...Pair) Pairs {
-	merge := append(slices.Clone(p), pairs...)
-	var filterInt int
-	for x := len(p); x < len(merge); x++ {
-		if !merge[:len(p)+filterInt].Contains(merge[x], true) {
-			merge[len(p)+filterInt] = merge[x]
-			filterInt++
+	p = slices.Clone(p)
```

Comment:
```suggestion
	c := slices.Clone(p)
```
Has no impact on allocs or performance. Even though the function doesn't have a pointer receiver, why allow the idea that you're doing anything with `p` when `p` isn't `p` and has no impact on `p`?
I know, I know, just `git gud`, but also removes two of three yellow squiggles

Reply:
My only argument against this, in general, is the bear trap:
When we switch to using `c` as suggested ( Or `n`-ew is what I'd normally do) we run the risk of people making future mistakes by using `p` instead of `c`.

What's "three yellow squiggles" you mention?

Reply:
I generally view non-pointer receivers as something you read from, since you can't modify the underlying, you are modifying a fake `p`, so your other foot gets in the bear trap, where you can think you are modifying `p` when you're not.
2 squiggles are `Assignment to the method receiver doesn't propagate to other calls ` and the other is the usual complaint about having struct functions with pointer and non-pointer receivers which go docs don't like, but I don't mind

Reply:
Swayed 👍 

Fixed 10ce34b7